### PR TITLE
Json editor for eval admin

### LINKED ISSUE #390: Evaluation Service - Update the admin editor for saved evaluations to show a CodeMirror json editor instead of just a text field
The editor for evaluations in the admin views doesn't have a proper json editor - only a textarea with escaped JSON. An overridden screen for that needs to be added that will apply a codemirror editor at that textarea like in the other editing views. An `onkeypress` event handler should be set on the editor to update the underlying textarea so all changes made in the editor end up in the textarea for submission.

### PR DESCRIPTION
**Pull request by christophertubbs**
Created the functionality used to view and edit json data in several views on the evaluation service admin site

closes #390

### CODE REVIEW

**Review by aaraney — CHANGES_REQUESTED** [review]
A few minor changes and we should be good to go! Thanks, @christophertubbs!

File: python/lib/evaluations/dmod/test/specification/test_deserialization.py [hunk 1]
```diff
@@ -53,6 +50,9 @@ def test_multitemplate(self):
         self.assertEqual(single_instance, pure_template_instance)
 
     def test_evaluation_deserialization(self):
+        from pprint import pprint
+
+        pprint(dir(specification.EvaluationSpecification))
```
[the remark this answers is not in the document]
Comment (christophertubbs) [reply python/lib/evaluations/dmod/test/specification/test_deserialization.py]: ... you weren't supposed to see that. Removed.

File: python/services/evaluationservice/dmod/evaluationservice/evaluation_service/forms/json.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+from dmod.evaluations import specification
+from dmod.evaluations.specification.base import get_subclasses
+
+from evaluation_service import models
+
+BINARY_TYPES = re.compile(
+    r'((?<!,)\s*\{\s*"type": "string",\s*"format": "binary"\s*},\s*|,\s*\{\s*"type": "string",\s*"format": "binary"\s*}\s*)'
+)
+"""
+A regular expression that finds all type definitions that are strings formatted as bytes
```
[the remark this answers is not in the document]
Comment (christophertubbs) [reply python/services/evaluationservice/dmod/evaluationservice/evaluation_service/forms/json.py:25]: Just added a details explanation on what strings will be found with the regex, what the regex is used for, and why the two extremely similar forms are necessary.

**Review by aaraney — APPROVED** [review]
Thanks for working through this with me, @christophertubbs!

**Pull request merged**